core/watcher/watchertest/watcher.go

```diff
@@ -138,7 +138,18 @@ func (w *WatcherC[T]) Check(assertion WatcherAssert[T]) {
 		select {
 		case actual, ok := <-w.Watcher.Changes():
 			w.c.Logf("WatcherC Watcher.Changes() => %# v", actual)
-			w.c.Assert(ok, jc.IsTrue)
+			if !ok {
+				wait := make(chan error)
+				go func() {
+					wait <- w.Watcher.Wait()
+				}()
+				select {
+				case <-time.After(testing.LongWait):
+					w.c.Fatalf("watcher never stopped")
+				case err := <-wait:
+					w.c.Fatalf("watcher killed with err: %q", err.Error())
+				}
```

Comment:
```suggestion
			if !ok {
				select {
				case <-time.After(testing.LongWait):
					w.c.Fatalf("watcher never stopped")
				case err := <-w.Watcher.Wait():
					w.c.Fatalf("watcher killed with err: %q", err.Error())
				}
```

Reply:
That won't work actually because `Wait()` doesn't return a channel, it just blocks, so we need to have a goroutine to fill the channel and select it like I did.

@manadart the reason I did this is not to wait forever until the worker returns, this way I can use the `testing.Longwait` instead.